src/main/java/org/gridsuite/modification/server/modifications/AbstractBranchModification.java

```diff
@@ -0,0 +1,169 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.gridsuite.modification.server.modifications;
+
+import com.powsybl.commons.reporter.Report;
+import com.powsybl.commons.reporter.Reporter;
+import com.powsybl.commons.reporter.TypedValue;
+import com.powsybl.iidm.network.Branch;
+import com.powsybl.iidm.network.CurrentLimits;
+import com.powsybl.iidm.network.CurrentLimitsAdder;
+import com.powsybl.iidm.network.LoadingLimits;
+import org.gridsuite.modification.server.dto.BranchModificationInfos;
+import org.gridsuite.modification.server.dto.CurrentLimitsModificationInfos;
+import org.gridsuite.modification.server.dto.CurrentTemporaryLimitModificationInfos;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author Florent MILLOT <florent.millot at rte-france.com>
+ */
+public abstract class AbstractBranchModification extends AbstractModification {
+
+    private static final String DURATION = "duration";
+    private static final String NAME = "name";
+    protected final BranchModificationInfos modificationInfos;
+
+    public AbstractBranchModification(BranchModificationInfos modificationInfos) {
+        this.modificationInfos = modificationInfos;
+    }
+
+    protected void modifyBranch(Branch branch, BranchModificationInfos branchModificationInfos, Reporter subReporter, String reporterKey, String reporterDefaultMessage) {
+
+        subReporter.report(Report.builder()
+                .withKey(reporterKey)
+                .withDefaultMessage(reporterDefaultMessage)
+                .withValue("id", branchModificationInfos.getEquipmentId())
+                .withSeverity(TypedValue.INFO_SEVERITY)
+                .build());
+        if (branchModificationInfos.getEquipmentName() != null) {
+            subReporter.report(ModificationUtils.getInstance().buildModificationReportWithIndentation(branch.getNameOrId(),
+                    branchModificationInfos.getEquipmentName().getValue(), "Name", 0));
+            branch.setName(branchModificationInfos.getEquipmentName().getValue());
+        }
+
+        if (characteristicsModified(branchModificationInfos)) {
+            modifyCharacteristics(branch, branchModificationInfos, subReporter);
+        }
+
+        CurrentLimitsModificationInfos currentLimitsInfos1 = modificationInfos.getCurrentLimits1();
+        CurrentLimitsModificationInfos currentLimitsInfos2 = modificationInfos.getCurrentLimits2();
+        List<Report> side1LimitsReports = new ArrayList<>();
+        //TODO FM why need to cast ???
```

Comment:
Anyone knows why I have to convert this ?

Reply:
Because `branch.getCurrentLimits1().orElse(null)` returns an `Object` that you need to cast to `CurrentLimits`

Reply:
Yes but actually it's because I had the specify the generic of Branch, otherwise the type is not working correctly

Reply:
with that : 

Optional<CurrentLimits> optCurrentLimits1 = branch.getCurrentLimits1();
CurrentLimits currentLimits1 = optCurrentLimits1.orElse(null);

there is no need to cast ...